core/services/offchainreporting/config_overrider.go

```diff
@@ -0,0 +1,129 @@
+package offchainreporting
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting/types"
+	"github.com/tevino/abool"
+)
+
+type ConfigOverriderImpl struct {
+	utils.StartStopOnce
+	logger *logger.Logger
+
+	flags           *ContractFlags
+	contractAddress ethkey.EIP55Address
+
+	flagsUpdateInterval time.Duration
+	isHibernating       *abool.AtomicBool
```

Comment:
atomic seems a bit low-level for this use case, see e.g. the package docs  ...
> Package atomic provides low-level atomic memory primitives useful for implementing synchronization algorithms.
> These functions require great care to be used correctly. Except for special, low-level applications, synchronization is better done with channels or the facilities of the sync package. Share memory by communicating; don’t communicate by sharing memory.

… and mailing lists threads such as this: https://groups.google.com/g/golang-nuts/c/0uPDCRiBWqc

How about a plain old mutex?

Reply:
oh interesting
```
The behaviour of mutexes is guaranteed by the Go memory model
(http://golang.org/ref/mem). The behaviour of the sync/atomic
functions is not.
```
we do use `sync/atomic` in quite a few places in the node, so might require further attention - thanks!